src/analysis/rust_type.rs

```diff
@@ -78,14 +79,8 @@ fn rust_type_full(env: &Env, type_id: library::TypeId, nullable: Nullable, by_re
     if by_ref {
         rust_type = rust_type.map_any(|s| format!("&{}", s));
     }
-    if *nullable {
-        // ignore nullability of containers, they don't require wrapping in an Option
-        match TypeKind::of(&env.library, type_id) {
-            TypeKind::Pointer |
-                TypeKind::Object |
-                TypeKind::Interface => rust_type = rust_type.map_any(|s| format!("Option<{}>", s)),
-            _ => (),
-        }
+    if *nullable && !skip_option {
+        rust_type = rust_type.map_any(|s| format!("Option<{}>", s));
```

Comment:
I'm starting to recall that some scalar typed parapeters might be erroneously marked as nullable (e.g. an outparam is nullable instead of optional) so the check for `ConversionType::Pointer` might indeed be needed. I guess I should've put a better justification in that commit.